packages/headless/esbuild.mjs

```diff
@@ -2,6 +2,7 @@ import {build} from 'esbuild';
 import alias from 'esbuild-plugin-alias';
 import {readFileSync, promises} from 'node:fs';
 import {resolve} from 'node:path';
+import resolveModule from 'resolve';
```

Comment:
When the package version was being bumped by `npm version`, the generated `package-lock` would sometimes hoist/unhoist certain packages, which caused `coveo.analytics` to move from the root `node_modules` to the Headless' `node_modules` and back randomly.

I found this simple fix, which isn't necessary anymore but feels cleaner than directly referencing `node_modules`, therefore I'm keeping it.

Reply:
I reckon you should be able to manage with just good ol' `require`
Do something like
```ts
import {createRequire} from 'node:module';
import {dirname, resolve} from 'node:path';
const require = createRequire(import.meta.url);

// ...
'coveo.analytics': (() => {
  const moduleDir = dirname(require.resolve('coveo.analytics.js/package.json'));
  const {module} = require('coveo.analytics.js/package.json');
  return resolve(moduleDir, module);
})()

Reply:
TIL we can use `require` in ESM like that. Will do.